src/transformers/tokenization_utils_base.py

```diff
@@ -2633,13 +2634,19 @@ def pad(
         if isinstance(encoded_inputs, (list, tuple)) and isinstance(encoded_inputs[0], (dict, BatchEncoding)):
             encoded_inputs = {key: [example[key] for example in encoded_inputs] for key in encoded_inputs[0].keys()}
 
-        assert "input_ids" in encoded_inputs, (
-            "You should supply an encoding or a list of encodings to this method. "
-            "An encoding is the output of one the encoding methods of the tokenizer, i.e. "
-            "__call__/encode_plus/batch_encode_plus. "
-        )
+        # ordered filtering of passed encoded_inputs keys
```

Comment:
@LysandreJik @sgugger @thomwolf - it would be great if you could check this change to allow `self.pad` to be used with Encoding Inputs that do not contain `input_ids`.